### ISSUE TITLE
Clutter false positive

### ISSUE THREAD

**Issue by josephzidell** [golangci/golangci-lint#1361]
Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.31.0 built from 3d6d0e7 on 2020-09-07T15:14:41Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
---
run:
  timeout: 3m
linters-settings:
  gocritic:
    disabled-checks:
      - captLocal
      - typeSwitchVar
  gofumpt:
    extra-rules: true
  misspell:
    locale: US
  nakedret:
    max-func-lines: 60
linters:
  enable:
    - asciicheck
    - bodyclose
    - deadcode
    - depguard
    - dogsled
    - dupl
    - errcheck
    - exhaustive
    - exportloopref
    - gocognit
    - goconst
    - gocritic
    - gocyclo
    - godot
    - godox
    - gofmt
    - gofumpt
    - goimports
    - golint
    - gomnd
    - gosec
    - gosimple
    - govet
    - ineffassign
    - misspell
    - nakedret
    - nlreturn
    - noctx
    - nolintlint
    - prealloc
    - scopelint
    - sqlclosecheck
    - staticcheck
    - structcheck
    - stylecheck
    - typecheck
    - unconvert
    - unused
    - varcheck
    - whitespace
    - wsl
issues:
  exclude-rules:
    - path: <redacted>.go
      linters:
        - gosimple
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.15 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/<user>/Library/Caches/go-build"
GOENV="/Users/<user>/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/<user>/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/<user>/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/9m/7mftnprj0t52fnhp1y56nfsm0000gp/T/go-build193552438=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ <redacted...>] 
INFO [config_reader] Used config file ../../.golangci.yml 
INFO [lintersdb] Active 41 linters: [asciicheck bodyclose deadcode depguard dogsled dupl errcheck exhaustive exportloopref gocognit goconst gocritic gocyclo godot godox gofmt gofumpt goimports golint gomnd gosec gosimple govet ineffassign misspell nakedret nlreturn noctx nolintlint prealloc scopelint sqlclosecheck staticcheck structcheck stylecheck typecheck unconvert unused varcheck whitespace wsl] 
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|files|imports|exports_file|name|types_sizes) took 403.05256ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 3.170302ms 
INFO [linters context/goanalysis] analyzers took 5.590834224s with top 10 stages: buildir: 4.09112558s, exhaustive: 318.950086ms, inspect: 227.919717ms, fact_deprecated: 217.233467ms, ctrlflow: 189.707728ms, printf: 172.508073ms, fact_purity: 158.829559ms, buildssa: 48.488422ms, gofumpt: 20.534131ms, goimports: 16.417968ms 
INFO [linters context/goanalysis] analyzers took 67.250933ms with top 10 stages: buildir: 61.656015ms, U1000: 5.594918ms 
INFO [runner] Issues before processing: 87, after processing: 2 
INFO [runner] Processors filtering stat (out/in): exclude: 2/87, max_per_file_from_linter: 2/2, max_same_issues: 2/2, path_shortener: 2/2, cgo: 87/87, path_prettifier: 87/87, skip_dirs: 87/87, exclude-rules: 2/2, uniq_by_line: 2/2, nolint: 2/2, diff: 2/2, max_from_linter: 2/2, severity-rules: 2/2, path_prefixer: 2/2, sort_results: 2/2, filename_unadjuster: 87/87, skip_files: 87/87, autogenerated_exclude: 87/87, identifier_marker: 87/87, source_code: 2/2 
INFO [runner] processing took 3.296436ms with stages: exclude: 1.265901ms, identifier_marker: 856.371µs, nolint: 621.031µs, path_prettifier: 223.654µs, cgo: 171.755µs, autogenerated_exclude: 79.565µs, source_code: 46.926µs, skip_dirs: 17.609µs, filename_unadjuster: 4.689µs, max_same_issues: 2.246µs, uniq_by_line: 1.789µs, path_shortener: 1.445µs, max_from_linter: 974ns, exclude-rules: 950ns, max_per_file_from_linter: 485ns, skip_files: 250ns, diff: 247ns, severity-rules: 208ns, sort_results: 204ns, path_prefixer: 137ns 
INFO [runner] linters took 2.732970101s with stages: goanalysis_metalinter: 2.592233787s, unused: 132.311373ms 
<file>.go:184:3: exitAfterDefer: log.Fatalln clutters `defer wg.Done()` (gocritic)
                log.Fatalln(..., err)
                ^
<file>.go:283:3: exitAfterDefer: log.Fatalf clutters `defer wg.Done()` (gocritic)
                log.Fatalf("failed to create file %s, %v", saveFileLocation, err)
                ^
INFO File cache stats: 3 entries of total size 45.2KiB 
INFO Memory: 33 samples, avg is 279.3MB, max is 409.5MB 
INFO Execution took 3.199977538s
```

</details>

<details><summary>Code</summary>

```go
	...
	// Signal completion to WaitGroup when finished
	defer wg.Done() // <-- here's the defer

	set, err := f(...)
	if err != nil {
		errs <- fmt.Sprintln(..., err)

		log.Fatalln(..., err) // <-- error reported on this line. this is line 184
	}
	...


// later in the file...

	// Signal completion to WaitGroup when finished
	defer wg.Done() // <-- here's the defer

	saveFileLocation := fileSaveFolder + fileName
	// check if the file already exists
	exists, err := isFileExists(saveFileLocation)
	if err != nil {
		log.Println("Cannot access", saveFileLocation, ":", err)

		return
	}

	if exists {
		log.Println(saveFileLocation, "already exists.")

		return
	}

	log.Println(fileName, "not set, getting file from S3")

	// Create a file to write the S3 object contents to
	file, err := os.Create(saveFileLocation)
	if err != nil {
		log.Fatalf("failed to create file %s, %v", saveFileLocation, err) // <-- error reported on this line. this is line 283
	}
```

**Comment by josephzidell** [comment]
Closing in favor of go-critic/go-critic#972
